cmd/gpu-kubelet-plugin/driver.go

```diff
@@ -177,6 +202,68 @@ func (d *driver) nodeUnprepareResource(ctx context.Context, claimNs kubeletplugi
 	return nil
 }
 
+func (d *driver) deviceHealthEvents(ctx context.Context, nodeName string) {
+	klog.Info("Processing device health notifications")
+	for {
+		select {
+		case <-ctx.Done():
+			klog.V(6).Info("Stop processing device health notifications")
+			return
+		case device, ok := <-d.deviceHealthMonitor.Unhealthy():
```

Comment:
What do you think about wrapping the logic in this case in a function and then call that function here. I think that would greatly help readability. It may also help with calling `release()` more reliably (if needed -- see discussion [here](https://github.com/NVIDIA/k8s-dra-driver-gpu/pull/689/files#r2454590696)).